src/metatensor/models/utils/data/dataset.py

```diff
@@ -125,50 +74,29 @@ def get_all_targets(dataset: Dataset) -> List[str]:
     # Iterate over all single instances of the dataset:
     target_names = []
     for index in range(len(dataset)):
-        _, targets = dataset[index]
-        target_names += list(targets.keys())
+        sample = dataset[index]._asdict()  # NamedTuple -> dict
+        sample.pop("structure")  # structure not needed
+        target_names += list(sample.keys())
 
     # Remove duplicates:
     return list(set(target_names))
 
 
-def collate_fn(batch: List[Tuple[System, Dict[str, TensorMap]]]):
+def collate_fn(batch: List[NamedTuple]) -> Tuple[List, Dict[str, TensorMap]]:
```

Comment:
Can't we get rid of this wrapper as well? We have the full freedom to change the code.

Reply:
I think @frostedoyster wanted to return the targets in a dict - if no longer the case we can change it

Reply:
@PicoCentauri the issue with using the default collate function from metatensor-learn is that it returns a named tuple, while our models, in order to be torch-script-able and consistent with the `metatensor.torch.atomistic` format for atomistic simulations, return a `Dict[str, TensorMap]`. I think this 3-line wrapper to adapt the output of the Dataset/loader is worth it

Reply:
Hmm but why is it then this way in metatensor-learn? Shouldn't this be consistent? 

I am not generally against a wrapper but it looks a bit weird that we have to write wrappers for our own library function.

Reply:
I think `metatensor.learn` is supposed to be more flexible than `metatensor.torch.atomistic`, mainly due to the lack of assumptions about torchscriptability.

More importantly, I see that we have some conflicts now...

Reply:
Hm, okay. Then I am fine if we add a little test for this wrapper.